apps/internal/base/base.go

```diff
@@ -345,6 +350,24 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	if silent.Claims == "" {
 		ar, err = AuthResultFromStorage(storageTokenResponse)
 		if err == nil {
+			if shouldRefresh(storageTokenResponse.AccessToken.RefreshOn.T) {
+				if tr, er := b.Token.Credential(ctx, authParams, silent.Credential); er == nil {
+					return b.AuthResultFromToken(ctx, authParams, tr)
+				} else if callErr, ok := er.(*errors.CallErr); ok {
```

Comment:
The type assertion is probably okay but errors.As would be more robust. Also, you don't need the else, because you return when the preceding if evaluates true

Reply:
Updated